From 822105616db1577cb72ce899dd88a068617ea7d4 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:04 -0400
Subject: [usb] fs: Add ep_to_host_endpoint helper function

Message-id: <1410887463-20674-293-git-send-email-dzickus@redhat.com>
Patchwork-id: 95360
O-Subject: [RHEL7 PATCH 292/591] usbfs: Add ep_to_host_endpoint helper function
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 2fec32b06e374642802f7fb4f5350317cd14732b
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed Oct 9 17:19:30 2013 +0200

    usbfs: Add ep_to_host_endpoint helper function

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/devio.c b/drivers/usb/core/devio.c
index d7571a6..502974b 100644
--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -769,6 +769,15 @@ static int check_ctrlrecip(struct dev_state *ps, unsigned int requesttype,
  return ret;
 }
 
+static struct usb_host_endpoint *ep_to_host_endpoint(struct usb_device *dev,
+           unsigned char ep)
+{
+ if (ep & USB_ENDPOINT_DIR_MASK)
+  return dev->ep_in[ep & USB_ENDPOINT_NUMBER_MASK];
+ else
+  return dev->ep_out[ep & USB_ENDPOINT_NUMBER_MASK];
+}
+
 static int match_devt(struct device *dev, void *data)
 {
  return dev->devt == (dev_t) (unsigned long) data;
@@ -1230,15 +1239,10 @@ static int proc_do_submiturb(struct dev_state *ps, struct usbdevfs_urb *uurb,
   if (ret)
    return ret;
  }
- if ((uurb->endpoint & USB_ENDPOINT_DIR_MASK) != 0) {
-  is_in = 1;
-  ep = ps->dev->ep_in[uurb->endpoint & USB_ENDPOINT_NUMBER_MASK];
- } else {
-  is_in = 0;
-  ep = ps->dev->ep_out[uurb->endpoint & USB_ENDPOINT_NUMBER_MASK];
- }
+ ep = ep_to_host_endpoint(ps->dev, uurb->endpoint);
  if (!ep)
   return -ENOENT;
+ is_in = (uurb->endpoint & USB_ENDPOINT_DIR_MASK) != 0;
 
  u = 0;
  switch(uurb->type) {
-- 
1.7.1